### PR TITLE
Feat/edit comment

### PR DESCRIPTION
**Pull request by lukbar48**

### CODE REVIEW

File: src/data-access/mutations/useAddReactionsComment.ts [hunk 1]
```diff
@@ -59,3 +56,45 @@ export const useAddReaction = () =>
       console.log('Error while adding reaction: ', err.message)
     },
   })
+
+const deleteComment = async (comment: EditComment): Promise<SubmitSuccess> => {
+  const { data } = await axios.delete(API.DELETE.deleteComment(comment), { data: comment })
+  return data
+}
```
Comment (iceu-bb) [inline src/data-access/mutations/useAddReactionsComment.ts]: I think you can only pass `commentId` as an argument here, don't need to pass the whole comment 🤔

File: src/data-access/mutations/useAddReactionsComment.ts [hunk 2]
```diff
@@ -59,3 +56,45 @@ export const useAddReaction = () =>
...
+export const useDeleteComment = () =>
+  useMutation<SubmitSuccess, AxiosError<{ errors: string[] }>, EditComment>(deleteComment, {
+    onSuccess: (payload) => {
+      queryClient.setQueryData<FeedPost[]>([QueryKeys.POSTS], (data) => {
+        if (!data) throw new Error('No posts found!')
+        const allPosts = data
+        const deletedCommentPostId = payload.post.id
+        const postIndex = allPosts?.findIndex((post) => post.id === deletedCommentPostId)
+        allPosts[postIndex] = payload.post
+        return allPosts
```
Comment (iceu-bb) [inline src/data-access/mutations/useAddReactionsComment.ts:73]: Is there a need to return `allPosts` from the delete hook?

File: src/screens/feed/Feed.tsx [hunk 3]
```diff
@@ -46,6 +69,59 @@ export const Feed = ({ route: { params: p } }: BottomTabNavigationProps<'FEED'>)
     return removeListener
   }, [navigation])
 
+  const openEditModal = (target: EditTargetType) => {
+    if (!(target.author === `${user?.firstName} ${user?.lastName}`)) return
```
Comment (iceu-bb) [inline src/screens/feed/Feed.tsx]: - Can we check the author by `id`? If the user will change the name this logic will be broken

File: src/screens/feed/Feed.tsx [hunk 4]
```diff
@@ -46,6 +69,59 @@ export const Feed = ({ route: { params: p } }: BottomTabNavigationProps<'FEED'>)
...
+    setEditTarget(target)
+    openModal?.()
+    if (target.type === 'comment') {
+      setEditingCommentTrue()
+    }
+  }
+
+  const handleModalDelete = () => {
```
Comment (iceu-bb) [inline src/screens/feed/Feed.tsx]:
```suggestion
  const onCommentDelete = () => {
```

File: src/screens/feed/components/CommentBox/CommentBox.tsx [hunk 5]
```diff
@@ -34,13 +50,19 @@ export const CommentBox = ({
...
+              id={comment.id}
+              key={comment.id}
             />
           ))
         ) : (
-          <Comment comment={comments[comments.length - 1]} />
+          <Comment
+            {...commonCommentProps}
+            comment={comments[comments.length - 1]}
+            id={comments[comments.length - 1].id}
```
Comment (iceu-bb) [inline src/screens/feed/components/CommentBox/CommentBox.tsx]: `comments[comments.length - 1]` can be declared as variable

**Review by iceu-bb — CHANGES_REQUESTED** [review]

File: src/data-access/mutations/useAddReactionsComment.ts (hunk 1 above)
Comment (lukbar48) [reply src/data-access/mutations/useAddReactionsComment.ts:63]: there is passed commentId and postID as object in data

File: src/screens/feed/Feed.tsx [hunk 6]
```diff
@@ -1,32 +1,108 @@
...
+  const { t } = useTranslation('feed')
+  const { user } = useUserContext()
   const flatListRef = useRef<FlatList | null>(null)
   const scrollRetries = useRef(0)
+
+  const [messageInputOpened, { setTrue: showMessageInput, setFalse: hideMessageInput }] =
+    useBooleanState(false)
+  const [isModalOpen, { setFalse: closeModal, setTrue: openModal }] = useBooleanState(false)
+  const [isEditingComment, { setFalse: setEditingCommentFalse, setTrue: setEditingCommentTrue }] =
+    useBooleanState(false)
```
Comment (iceu-bb) [inline src/screens/feed/Feed.tsx]: I think you may not need `isEditingComment` state as you can check

`if(editTarget.type === "comment")`

File: src/screens/feed/Feed.tsx [hunk 7]
```diff
@@ -1,32 +1,108 @@
...
   const navigation = useNavigation()
-
+  const { t } = useTranslation('feed')
+  const { user } = useUserContext()
   const flatListRef = useRef<FlatList | null>(null)
   const scrollRetries = useRef(0)
+
+  const [messageInputOpened, { setTrue: showMessageInput, setFalse: hideMessageInput }] =
+    useBooleanState(false)
+  const [isModalOpen, { setFalse: closeModal, setTrue: openModal }] = useBooleanState(false)
```
Comment (iceu-bb) [inline src/screens/feed/Feed.tsx]:
```suggestion
  const [isOptionsModalOpen, { setFalse: closeOptionsModal, setTrue: openOptionsModal }] = useBooleanState(false)
```

File: src/screens/feed/Feed.tsx [hunk 8]
```diff
@@ -1,32 +1,108 @@
...
+  const [isEditingComment, { setFalse: setEditingCommentFalse, setTrue: setEditingCommentTrue }] =
+    useBooleanState(false)
+  const [editTarget, setEditTarget] = useState<EditTargetType>()
+  const [messageContent, setMessageContent] = useState('')
```
Comment (iceu-bb) [inline src/screens/feed/Feed.tsx]: I wonder if you can derive `messageContent` from `editTarget.text` and use it as a source of truth ?

File: src/screens/feed/Feed.tsx [hunk 9]
```diff
@@ -1,32 +1,108 @@
...
+    }
+  }
+
+  const onCommentEdit = () => {
+    hideMessageInput()
+    if (editTarget?.type === 'comment') {
+      editComment({
+        ...editTarget,
+        text: messageContent,
+      })
```
Comment (iceu-bb) [inline src/screens/feed/Feed.tsx]:
```suggestion
      editComment({ ...editTarget, text: messageContent })
```

**Review by iceu-bb — CHANGES_REQUESTED** [review]

**Review by iceu-bb — APPROVED** [review]

**Pull request merged**